README.md

```diff
@@ -81,83 +75,179 @@ Install-Package Winton.AspNetCore.Seo
+The robots.txt file can be configured in the same manner as the sitemap.
+If `AddSeoWithDefaultRobots` is called the default robots.txt options will be used which applies the following policy:
 
-* The URL of the sitemap is specified in all environments.
-* In the production environment all user agents are allowed to index any part of the site.
-* In all other environments, all user agents are disallowed from indexing any part of the site.
+-   The URL of the sitemap is specified in all environments.
+-   In the production environment all user agents are allowed to index any part of the site.
+-   In all other environments, all user agents are disallowed from indexing any part of the site.
 
-*Note: It's always worth remembering that malicious bots can just ignore a robots.txt file, so it should not be relied on to hide any sensitive parts of the site.*
+_Note: It's always worth remembering that malicious bots can just ignore a robots.txt file, so it should not be relied on to hide any sensitive parts of the site._
 
-To provide your own implementation you can do the following:
+Calling `AddSeo` instead allows you to specify the options. There are several ways they can be specified:
 
-```csharp
-public void ConfigureServices(IServiceCollection services)
-{
-    // Register your implementation of IRobotsTxtOptions first
-    services.AddSingleton<IRobotsTxtOptions, YourRobotsTxtOptions>();
-    services.AddSeo(sitemapConfig);
-}
-```
+1. Using a delegate.
+   For example:
+
+    ```csharp
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddSeo(
+            options =>
+            {
+                options.RobotsTxt.AddSitemapUrl = false;
+                options.RobotsTxt.UserAgentRecords = new List<UserAgentRecord>
+                {
+                    new UserAgentRecord
+                    {
+                        Disallow = new List<string> { "/login" },
+                        UserAgent = "bing"
+                    }
+                };
+            });
+    }
+    ```
+
+    This is useful when you want to define your the robots.txt using code.
+    You can also configure the sitemap in the same delegate.
+    There is also a version of the delegate that has access to `IWebHostEnvironment` for when the configuration is dependant on the environment.
+
+1. From config.
+   For example:
+
+    ```csharp
+    services.AddSeo(_configuration);
+    ```
+
+    This is useful when you want to define the robots.txt in config, such as `appsettings.json`.
+
+    The config must contain a section with the key `"Seo"`, for example:
+
+    ```json
+    {
+        "Seo": {
+            "RobotsTxt": {
+                "AddSitemapUrl": false,
+                "UserAgentRecords": [
+                    {
+                        "UserAgent": "bing",
+                        "Disallow": "/login" -- should be an array
```

Comment:
Looks like GitHub doesn't like this syntax - what does the comment mean?

Reply:
Ah, I need to fix that, good spot